mono/metadata/w32file-unity.c

```diff
@@ -1,282 +1,529 @@
+{
+	return;
+}
+#endif /* HOST_WIN32 */
+
+gpointer
+mono_w32file_create(const gunichar2 *name, guint32 fileaccess, guint32 sharemode, guint32 createmode, guint32 attrs)
+{
+	int error = 0;
+	gpointer handle;
+	gchar* palPath = mono_unicode_to_external(name);
+	handle =  UnityPalOpen(palPath, (int) createmode, (int) fileaccess, (int) sharemode, attrs, &error);
+	mono_w32error_set_last(error);
+	g_free(palPath);
+
+	if (handle == NULL)
+		return INVALID_HANDLE_VALUE;
+
+	return handle;
 }
 
 gboolean
-mono_w32file_delete (const gunichar2 *name)
+mono_w32file_close (gpointer handle)
 {
-	g_assert(0 && "This function is not yet implemented for the Unity platform.");
-	return FALSE;
+	if (handle == NULL)
+		return FALSE;
+	
+	int error = 0;
+	gboolean result = UnityPalClose(handle, &error);
+	mono_w32error_set_last(error);
+	
+	return result;
 }
 
 gboolean
 mono_w32file_read(gpointer handle, gpointer buffer, guint32 numbytes, guint32 *bytesread)
 {
-	g_assert(0 && "This function is not yet implemented for the Unity platform.");
-	return FALSE;
+	int error = 0;
+	
+	*bytesread =  UnityPalRead(handle, buffer, numbytes, &error);
+	mono_w32error_set_last(error);
+	
+	return TRUE;
 }
 
 gboolean
 mono_w32file_write (gpointer handle, gconstpointer buffer, guint32 numbytes, guint32 *byteswritten)
 {
-	g_assert(0 && "This function is not yet implemented for the Unity platform.");
-	return FALSE;
+	int error = 0;
+
+	*byteswritten = UnityPalWrite(handle, buffer, numbytes, &error);
+	mono_w32error_set_last(error);
+
+	return (*byteswritten > 0);
 }
 
 gboolean
 mono_w32file_flush (gpointer handle)
 {
-	g_assert(0 && "This function is not yet implemented for the Unity platform.");
-	return FALSE;
+	int error = 0;
+	
+	gboolean result = UnityPalFlush(handle, &error);
+	mono_w32error_set_last(error);
+	
+	return result;
 }
 
 gboolean
 mono_w32file_truncate (gpointer handle)
 {
-	g_assert(0 && "This function is not yet implemented for the Unity platform.");
-	return FALSE;
+	int error = 0;
+
+	gboolean result = UnityPalTruncate(handle, &error);
+	mono_w32error_set_last(error);
+
+	return result;
 }
 
 guint32
 mono_w32file_seek (gpointer handle, gint32 movedistance, gint32 *highmovedistance, guint32 method)
 {
-	g_assert(0 && "This function is not yet implemented for the Unity platform.");
-	return 0;
+	int error = 0;
+	
+	int32_t result = UnityPalSeek(handle, movedistance, 0, &error);
+	mono_w32error_set_last(error);
+
+	return result;
 }
 
 gint
 mono_w32file_get_type (gpointer handle)
 {
-	g_assert(0 && "This function is not yet implemented for the Unity platform.");
-	return 0;
+	if (handle == NULL)
+		return 0;
+
+	return UnityPalGetFileType(handle);
 }
 
 gboolean
 mono_w32file_get_times (gpointer handle, FILETIME *create_time, FILETIME *access_time, FILETIME *write_time)
 {
-	g_assert(0 && "This function is not yet implemented for the Unity platform.");
-	return FALSE;
+	/* Not Supported in UnityPAL */
+	g_assert_not_reached();
```

Comment:
We should standardize a not support assert, I think. I was using this for the event code:

```
g_assertion_message("Named events are not supported by the Unity platform.");
```

I'm not sure if that is the best option though. Any other ideas?